### ISSUE TITLE
False-positive warning when @Nullable is casted to a @NonNull type

### ISSUE THREAD

**Issue by pfink** [openhab/static-code-analysis#431]
If I cast a `@Nullable` type to a `@NonNull` type, a false-positive warning is thrown.

Example:
```
public void mymethod(@Nullable Object myobject) {
        if (myobject != null) {
            this.mysecondmethod((@NonNull Object) myobject);
        }
 }
```

There is no problem with this code, but currently it will wrongly throw the warning: `There is no need for a @NonNull annotation because it is set as default. Only @Nullable should be used`

**Comment by wborn** [comment]
When I hit save on the flicbutton files my Eclipse auto save actions removed all those unneccessary casts automatically.
The only issue is that you must assign `@Nullable` fields to a local variable. That's because another thread can change the value of a field at any time to `null`. So if you annotate fields as `@Nullable` they can never be inferred as `@NonNull`, unless you assign them to a local variable.

**Comment by pfink** [comment]
Ah okay, that makes sense. Probably I should use Eclipse at least when I do bigger PRs next time ^^
